autonat/README.md

```diff
@@ -78,6 +78,12 @@ If all dials fail, the receiver sends a `DialResponse` message with the
 successfully, it sends a `DialResponse` with the `ResponseStatus` `OK`. It
 SHOULD include the address it successfully dialed in its response.
 
+If the receiver elects not to dial any addresses from the message - for example
+because they are invalid, because they resolve to the same host as the receiver,
+because they only include private addresses, because it does not support the
+relevant transports, or because it has no capacity, is sends a `DialResponse`
```

Comment:
```suggestion
relevant transports, or because it has no capacity, it SHOULD send a `DialResponse`
```

> We should probably add that you can’t trust on your peer doing this correctly - the Go implementation has been sending the wrong error code for a long time (until Elena fixed it recently).

@marten-seemann What do you think about using the keyword _SHOULD_, in the sense that it is recommended but one can not trust that the other peer actually does it?

Reply:
Technically, this is a MUST. You _should_ be able to trust your peer's reporting here.

In practice you can't. But that's just because the Go implementation was just not compliant with the spec, which brought us to the mess we're in right now (one more reason for Autonat v2!). I suggest adding a note here that this was the case.